src/blog/2019-02-28-learning-with-dockter/index.md

```diff
@@ -0,0 +1,31 @@
+---
+extends: blog/_post.html
+title: Learning Docker through Dockter
+author: Aleksandra Pawlik
+date: 28 February 2019
+abstract: |
+    This is a short summary of my, somewhat painful, route to discovery that what I thought I knew about Docker was in fact just the tip of the iceberg. Fortunately to me, working in Stencila I upgraded my knowledge through testing and using Dockter, a container image builder for researchers.
+forum_topic:
+image: docker-error.png
+---
+
+This is a short summary of my, somewhat painful, route to discovery that what I thought I knew about Docker was in fact just the tip of the iceberg. Fortunately to me, working in Stencila I upgraded my knowledge through testing and using [Dockter](https://stencila.github.io/dockter/),  a container image builder for researchers.
+
+I used Docker for the first time several years ago when it entered the world of computational research. Working with computational research communities I was well aware of the problems with reproducibility due to dependency hell. Docker images appeared to be (and are!) a great solution. However, not actually doing
+computational research myself, I only used Docker a couple of times following a tutorial to build an image just to see how it would go. It went all fine, as usually well prepared tutorials present the user with a typical and uncomplicated case. I have run several Docker images that I got from the depot. That has also worked fine for me. And that would be it when it came to my hands-on experience with Docker. 
+
+
+![docker-error.png](docker-error.png){style="width: 80%;" }
+
+Hence I was convinced that Docker is the silver bullet for the reproducibility problems in computational research. I happily recommended Docker to a number of researchers that I talked to and even to some developers who were struggling with dependency hell. It was only after I started helping with Stencila's Dockter that I understood how naive I was.
+
+First thing first, I didn't realize that rebuilding images with Docker can take a long time even if you only changed a tiny bit in your Dockerfile. In most cases, it is a recommended practice to create Docker images with as few layers as possible. That means that making even small changes in listing required packages in Dockerfile results in Docker rebuilding the whole image. Dockter deals with that problem pretty well by rebuilding the images from the layer where changes occurred. 
```

Comment:
I'm not too familiar with the details of the Dockter workflow, but is it worth emphasizing that the change layer detection is automatic?
> "Dockter deals with that problem pretty well by **automatically** rebuilding the images **only** from the layer where changes occurred.